src/main/java/com/google/step/snippet/servlets/SearchServlet.java

```diff
@@ -137,6 +148,21 @@ public void doGet(HttpServletRequest request, HttpServletResponse response)
       allCards = Collections.emptyList();
     }
 
+    if (!preferredSite.isEmpty()) {
+      List<Card> sortedCards = new ArrayList<>(allCards);
```

Comment:
Do you need `sortedCards`? Can you just use `allCards`?

I saw in another comment you mentioned that doing so could result in a ConcurrentAccessException. CMIIW, but I think by not using the for-each loop, you won't run into this issue anymore?